### PR TITLE
GH-44619: [GLib] Add GArrowDecimal32Scalar

### PR DESCRIPTION
**Pull request by hiroyuki-sato**
### Rationale for this change

The `arrow::Decimal32Scalar` has been released.
GLib needs to implement `GArrowDecimal32Scalar`.

### What changes are included in this PR?

Implement `GArrowDecimal32Scalar`.

### Are these changes tested?

YES

### Are there any user-facing changes?

NO

* GitHub Issue: #44619

### CODE REVIEW

File: c_glib/arrow-glib/scalar.h [hunk 1]
```diff
@@ -501,19 +501,35 @@ GARROW_AVAILABLE_IN_8_0
 GArrowMonthDayNano *
 garrow_month_day_nano_interval_scalar_get_value(GArrowMonthDayNanoIntervalScalar *scalar);
 
+#define GARROW_TYPE_DECIMAL32_SCALAR (garrow_decimal32_scalar_get_type())
+GARROW_AVAILABLE_IN_5_0
```
Comment (kou) [inline c_glib/arrow-glib/scalar.h:505]:
```suggestion
GARROW_AVAILABLE_IN_19_0
```

File: c_glib/arrow-glib/scalar.h [hunk 2]
```diff
@@ -501,19 +501,35 @@ GARROW_AVAILABLE_IN_8_0
...
+G_DECLARE_DERIVABLE_TYPE(
+  GArrowDecimal32Scalar, garrow_decimal32_scalar, GARROW, DECIMAL32_SCALAR, GArrowScalar)
+struct _GArrowDecimal32ScalarClass
+{
+  GArrowScalarClass parent_class;
+};
+
+GARROW_AVAILABLE_IN_5_0
```
Comment (kou) [inline c_glib/arrow-glib/scalar.h:513]:
```suggestion
GARROW_AVAILABLE_IN_19_0
```

File: c_glib/arrow-glib/scalar.h [hunk 3]
```diff
@@ -501,19 +501,35 @@ GARROW_AVAILABLE_IN_8_0
...
+GArrowDecimal32Scalar *
+garrow_decimal32_scalar_new(GArrowDecimal32DataType *data_type, GArrowDecimal32 *value);
+GARROW_AVAILABLE_IN_5_0
```
Comment (kou) [inline c_glib/arrow-glib/scalar.h:516]:
```suggestion
GARROW_AVAILABLE_IN_19_0
```

File: c_glib/arrow-glib/scalar.h [hunk 4]
```diff
@@ -501,19 +501,35 @@ GARROW_AVAILABLE_IN_8_0
...
+GArrowDecimal32 *
+garrow_decimal32_scalar_get_value(GArrowDecimal32Scalar *scalar);
+
 #define GARROW_TYPE_DECIMAL64_SCALAR (garrow_decimal64_scalar_get_type())
-GARROW_AVAILABLE_IN_19_0
+GARROW_AVAILABLE_IN_5_0
```
Comment (kou) [inline c_glib/arrow-glib/scalar.h:521]:
```suggestion
GARROW_AVAILABLE_IN_19_0
```

File: c_glib/arrow-glib/scalar.h [hunk 5]
```diff
@@ -501,19 +501,35 @@ GARROW_AVAILABLE_IN_8_0
...
 G_DECLARE_DERIVABLE_TYPE(
   GArrowDecimal64Scalar, garrow_decimal64_scalar, GARROW, DECIMAL64_SCALAR, GArrowScalar)
 struct _GArrowDecimal64ScalarClass
 {
   GArrowScalarClass parent_class;
 };
 
-GARROW_AVAILABLE_IN_19_0
+GARROW_AVAILABLE_IN_5_0
```
Comment (kou) [inline c_glib/arrow-glib/scalar.h:529]:
```suggestion
GARROW_AVAILABLE_IN_19_0
```

File: c_glib/arrow-glib/scalar.h [hunk 6]
```diff
@@ -501,19 +501,35 @@ GARROW_AVAILABLE_IN_8_0
...
 GArrowDecimal64Scalar *
 garrow_decimal64_scalar_new(GArrowDecimal64DataType *data_type, GArrowDecimal64 *value);
-GARROW_AVAILABLE_IN_19_0
+GARROW_AVAILABLE_IN_5_0
```
Comment (kou) [inline c_glib/arrow-glib/scalar.h:532]:
```suggestion
GARROW_AVAILABLE_IN_19_0
```

**Comment by hiroyuki-sato** [comment]
@kou Thanks I've forgotten to fix the `GARROW_AVAILABLE` parts.

**Review by kou — APPROVED** [review]
+1

**Pull request merged**